Lib/multiprocessing/shared_memory.py

```diff
@@ -138,34 +130,99 @@ def unlink(self):
         pass
 
 
-class PosixSharedMemory(_PosixSharedMemory):
+# FreeBSD (and perhaps other BSDs) limit names to 14 characters.
+_SHM_SAFE_NAME_LENGTH = 14
```

Comment:
I don't know if this limit bothers anyone, but it's probably not relevant anymore. I wrote `posix_ipc` back when FreeBSD was still developing POSIX IPC support, so in the early days (~8-10 years ago) it was still a little rough. That's probably where this limit came from. It improved a lot between FreeBSD 7.2 and 8.0. I just tested FreeBSD 11.1 with my `posix_ipc` module and I was able to create a semaphore with a 91 character name, so somewhere along the line this limit became irrelevant, I just can't say when.

Reply:
I think I would prefer to err on the side of caution by keeping this for now and potentially relaxing it in the future.  I suspect there are limits on various platforms that still need to be respected but this limit was the most conservative I found documented.  It is perhaps overly optimistic to say so but I hope that this limit (as it stands) does not create any true hardships for users when they try to come up with unique names for things.

Reply:
FWIW this limit still exists and affects at least Linux, Solaris and AIX. I have code in psutil which takes this into account.